server/service/store/handler_store_data.go

```diff
@@ -0,0 +1,42 @@
+package store
+
+import (
+	"encoding/hex"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func HandleStoreData(w http.ResponseWriter, r *http.Request) {
+	dataHashStr := mux.Vars(r)["key"]
+	_, err := hex.DecodeString(dataHashStr)
+	if err != nil {
+		log.Errorf("failed to decode dataHash: %s", err.Error())
+		http.Error(w, "failed to decode dataHash", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		log.Errorf("failed to read data: %v", err.Error())
+		http.Error(w, "failed to read data", http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/octet-stream")
```

Comment:
I wonder that it need to be set a `Content-Type` like this.

Reply:
I think you can get `content-type` from request header. We don't have any values in the response body, so this doesn't seem to be a problem.